includes/classes/EnqueueScript.php

```diff
@@ -0,0 +1,284 @@
+<?php
+/**
+ * This class use to register script.
+ *
+ * This class handles:
+ *   - Script dependencies.
+ *     This uses asset information to set script dependencies,
+ *     and version generated by @wordpress/dependency-extraction-webpack-plugin package.
+ *   - Script localization.
+ *     It also handles script translation registration.
+ *
+ * @package distributor
+ * @since   x.x.x
+ */
+
+namespace Distributor;
+
+use Exception;
+
+/**
+ * Class EnqueueScript
+ *
+ * @since x.x.x
+ */
+class EnqueueScript {
+	/**
+	 * Script ID.
+	 *
+	 * @since x.x.x
+	 */
+	private string $script_id;
```

Comment:
```suggestion
	/**
	 * Script Handle.
	 *
	 * @since x.x.x
	 */
	private string $script_handle;
```

Let's call this `handle` to match the WP naming convention.

:1234: This will need to be renamed throughout.